vispy/plot/plot.py

```diff
@@ -32,7 +32,8 @@ def image(*args, **kwds):
     """
     canvas = SceneCanvas(keys='interactive')
     canvas.view = canvas.central_widget.add_view()
-    canvas.image = visuals.Image(*args, **kwds)
+    _args = (args[0].astype(np.ubyte),) + args[1:]
```

Comment:
I'm wondering whether this should be here of in `visuals.Image`.
Also it may not be correct to cast to ubyte: the user could provide a floating-point texture, in which case we need to cast from float64 to float32. @campagnola or @almarklein may have a better idea of how to implement this correctly.